src/ansi-c/ansi_c_entry_point.cpp

```diff
@@ -268,10 +268,11 @@ bool generate_ansi_c_start_function(
         argc_symbol.is_lvalue = true;
         argc_symbol.mode = ID_C;
 
-        if(!symbol_table.insert(std::move(argc_symbol)).second)
+        auto r = symbol_table.insert(argc_symbol);
+        if (!r.second && r.first != argc_symbol)
         {
           messaget message(message_handler);
-          message.error() << "failed to insert argc symbol" << messaget::eom;
+          message.error() << "argc already exists but is not usable" << messaget::eom;
```

Comment:
What does `is not usable` mean?

Reply:
It is not equal to the expected symbol used for `argc` or `argv`.  It might mean that it has an unexpected type, differing life-time, etc.  TBH I struggle to think how this could happen but @tautschnig wanted a test and a test without error handling seems ... stupid.

Hmmm... maybe if you had a version of `main` that did something unusual and very non-standard with it's arguments and then tried to set the entry point to `main`, maybe you could trigger this.